scipy/signal/signaltools.py

```diff
@@ -2252,36 +2233,66 @@ def resample(x, num, t=None, axis=0, window=None):
             W = window
         else:
             W = fftpack.ifftshift(get_window(window, Nx))
-        newshape = [1] * x.ndim
-        newshape[axis] = len(W)
-        W.shape = newshape
-        X = X * W
-        W.shape = (Nx,)
+        newshape_W = [1] * x.ndim
+        newshape_W[axis] = len(W)
     sl = [slice(None)] * x.ndim
     newshape = list(x.shape)
     newshape[axis] = num
     N = int(np.minimum(num, Nx))
-    Y = zeros(newshape, 'D')
-    sl[axis] = slice(0, (N + 1) // 2)
-    Y[tuple(sl)] = X[tuple(sl)]
-    sl[axis] = slice(-(N - 1) // 2, None)
-    Y[tuple(sl)] = X[tuple(sl)]
-
-    if N % 2 == 0:  # special treatment if low number of points is even. So far we have set Y[-N/2]=X[-N/2]
-        if N < Nx:  # if downsampling
-            sl[axis] = slice(N//2,N//2+1,None)  # select the component at frequency N/2
-            Y[tuple(sl)] += X[tuple(sl)]  # add the component of X at N/2
-        elif N < num:  # if upsampling
-            sl[axis] = slice(num-N//2,num-N//2+1,None)  # select the component at frequency -N/2
-            Y[tuple(sl)] /= 2  # halve the component at -N/2
-            temp = Y[tuple(sl)]
-            sl[axis] = slice(N//2,N//2+1,None)  # select the component at +N/2
-            Y[tuple(sl)] = temp  # set that equal to the component at -N/2
-
-    y = fftpack.ifft(Y, axis=axis) * (float(num) / float(Nx))
-
-    if x.dtype.char not in ['F', 'D']:
-        y = y.real
+
+    # Can we use faster real FFT?
```

Comment:
This might be more clear as "Check if we can use faster real FFT." I initially read this as a suggestion for future exploration rather than a code comment.